pulsar-broker/src/test/java/com/yahoo/pulsar/client/impl/BrokerConsumerStatsTest.java

```diff
@@ -0,0 +1,155 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yahoo.pulsar.client.impl;
+
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.yahoo.pulsar.broker.service.BrokerTestBase;
+import com.yahoo.pulsar.client.api.Consumer;
+import com.yahoo.pulsar.client.api.ConsumerConfiguration;
+import com.yahoo.pulsar.client.api.Message;
+import com.yahoo.pulsar.client.api.Producer;
+import com.yahoo.pulsar.client.api.ProducerConfiguration;
+import com.yahoo.pulsar.client.api.PulsarClientException;
+import com.yahoo.pulsar.client.api.SubscriptionType;
+
+public class BrokerConsumerStatsTest extends BrokerTestBase {
+    private static final Logger log = LoggerFactory.getLogger(BrokerConsumerStatsTest.class);
+    private final int totalMessages = 50;
+    private String topicName = "persistent://prop/cluster/ns/topic-";
+
+    @BeforeClass
+    @Override
+    public void setup() throws Exception {
+        baseSetup();
+    }
+
+    @AfterClass
+    @Override
+    protected void cleanup() throws Exception {
+        internalCleanup();
+    }
+
+    @Test()
+    public void testSharedSubscriptionMessageBacklog() throws PulsarClientException {
+        String topicNamePostFix = "testSharedSubscriptionMessageBacklog";
+        ConsumerConfiguration conf = new ConsumerConfiguration();
+        conf.setSubscriptionType(SubscriptionType.Shared);
+        Consumer consumer1 = pulsarClient.subscribe(topicName + topicNamePostFix, "my-subscriber-name", conf);
+        Consumer consumer2 = pulsarClient.subscribe(topicName + topicNamePostFix, "my-subscriber-name", conf);
+        Consumer consumer3 = pulsarClient.subscribe(topicName + topicNamePostFix, "my-subscriber-name", conf);
+
+        Producer producer = pulsarClient.createProducer(topicName + topicNamePostFix, new ProducerConfiguration());
+        for (int i = 0; i < totalMessages; i++) {
+            String message = "my-message-" + i;
+            producer.send(message.getBytes());
+        }
+
+        int count = 0;
+        try {
+            Message msg = null;
+            while (true) {
+                msg = consumer1.receive(10, TimeUnit.MILLISECONDS);
+                if (msg != null) {
+                    consumer1.acknowledge(msg);
+                    count++;
+                } else {
+                    break;
+                }
+            }
+
+            log.debug(consumer1.getBrokerConsumerStatsAsync().get().toString());
+            Assert.assertEquals(consumer1.getBrokerConsumerStatsAsync().get().getMsgBacklog(), totalMessages - count);
+
```

Comment:
Since it's shared subscription we send 20 messages per consumer in round robin fashion